public/app/plugins/datasource/tempo/traceql/traceql.ts

```diff
@@ -160,6 +171,33 @@ const language: languages.IMonarchLanguage = {
       [/(@digits)[lL]?/, 'number'],
     ],
 
+    withClause: [
+      // whitespace
+      [/\s+/, ''],
+      // parameter names like 'most_recent'
+      [/[a-zA-Z_][a-zA-Z0-9_]*/, {
+        cases: {
+          '@withParameters': 'variable',
+          '@default': 'identifier'
+        }
+      }],
+      // assignment operator
+      [/=/, 'delimiter'],
+      // boolean values
+      [/\b(true|false)\b/, 'type'],
+      // strings
+      [/"([^"\\]|\\.)*"/, 'string'],
+      [/'([^'\\]|\\.)*'/, 'string'],
+      // numbers
+      [/\d+(\.\d+)?/, 'number'],
+      // comma separator for multiple parameters
+      [/,/, 'delimiter'],
+      // closing parenthesis - return to root
+      [/\)/, { token: 'delimiter.bracket', next: '@pop' }],
+      // anything else
+      [/./, '']
+    ],
```

Comment:
Looks like you're making this a bit more robust / future proof than we need to. I'd prefer to keep this simple for now to support our use case and we can upgrade this later if/as the language evolves. 

Something like;

    [/\s+/, ''],                                    // whitespace
    [/most_recent/, 'variable'],                    // parameter name
    [/=/, 'delimiter'],                             // assignment operator
    [/\b(true|false|0|1)\b/, 'type'],               // boolean values
    [/\)/, { token: 'delimiter.bracket', next: '@pop' }], // closing parenthesis